docs/rules/semi-style.md

```diff
@@ -0,0 +1,105 @@
+# Enforce location of semicolons (semi-style)
+
+Generally, semicolons are at the end of lines. However, in semicolon-less style, semicolons are at the beginning of lines. This rule enforces that semicolons are at the configured location.
+
+## Rule Details
+
+This rule reports line terminators around semicolons.
+
+This rule has 2 options `statements` and `forLoopHead`.
+
+```json
+{
+    "semi-style": ["error", "last"],
+    // OR
+    "semi-style": ["error", {"statements": "last",  "forLoopHead": "last"}]
+}
+```
+
+- `statements` ... configuration of statements.
+- `forLoopHead` ... configuration of `for(;;){}`.
```

Comment:
Is it common for people to have a different preference for `for` loops than they would have for statements? I'm wondering if it's worth having a separate option for loops.